src/Confluent.Kafka/ProducerBuilder.cs

```diff
@@ -20,6 +20,40 @@
 
 namespace Confluent.Kafka
 {
+    /// <summary>
+    ///     Calculate a partition number given a <paramref name="topic" />,
+    ///     <paramref name="partitionCount" /> and serialized <paramref name="keyData" />.
+    /// </summary>
+    /// <remarks>
+    ///     A partioner instance may be called in any thread at any time and
+    ///     may be called multiple times for the same message/key.
+    ///
+    ///     A partitioner:
+    ///     - MUST NOT call any method on the producer instance.
+    ///     - MUST NOT block or execute for prolonged periods of time.
+    ///     - MUST return a value between 0 and partition_cnt-1, or the
+    ///       special <seealso cref="Confluent.Kafka.Partition.Any"/> value.
```

Comment:
What does that do?

Reply:
it returns -1, which I believe should send to a random partition (perhaps sticky). will add a note.

Reply:
huh, my assumption was wrong, removing this as an option. value can only be 0 .. partitionCnt - 1.

Reply:
we should perhaps support this in librdkafka.